airbyte-integrations/connectors/destination-elasticsearch-strict-encrypt/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchStrictEncryptDestination.java

```diff
@@ -38,4 +46,25 @@ public ConnectorSpecification modifySpec(ConnectorSpecification originalSpec) th
     return spec;
   }
 
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) throws Exception {
+
+    final ConnectorConfiguration configObject = convertConfig(config);
+    if (Objects.isNull(configObject.getEndpoint())) {
+      return new AirbyteConnectionStatus()
+          .withStatus(AirbyteConnectionStatus.Status.FAILED).withMessage("endpoint must not be empty");
+    }
+
+    if (new URL(configObject.getEndpoint()).getProtocol().equals(HTTP)) {
+      return new AirbyteConnectionStatus()
+          .withStatus(AirbyteConnectionStatus.Status.FAILED).withMessage("only https protocol is allowed");
```

Comment:
Can we have a more descriptive message that specifies that this is due to the `endpoint` being insecure? 

A message like `Server Endpoint requires HTTPS`

Reply:
Updated, thanks